pkg/api/deploymentapi/update_payload.go

```diff
@@ -0,0 +1,219 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentapi
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+)
+
+// NewUpdateRequest generates a DeploymentUpdateRequest from a GetResponse.
+func NewUpdateRequest(res *models.DeploymentGetResponse) *models.DeploymentUpdateRequest {
+	if res == nil {
+		return nil
+	}
+
+	var esRefID string
+	var req = models.DeploymentUpdateRequest{
+		Name:         *res.Name,
+		Resources:    &models.DeploymentUpdateResources{},
+		PruneOrphans: ec.Bool(false),
+	}
+
+	for _, r := range res.Resources.Elasticsearch {
+		if resource := parseElasticsearchGetResponse(r, &esRefID); resource != nil {
+			req.Resources.Elasticsearch = append(req.Resources.Elasticsearch,
+				resource,
+			)
+		}
+	}
+
+	for _, r := range res.Resources.Kibana {
+		if resource := parseKibanaGetResponse(r, esRefID); resource != nil {
+			req.Resources.Kibana = append(req.Resources.Kibana,
+				resource,
+			)
+		}
+	}
+
+	for _, r := range res.Resources.Apm {
+		if resource := parseApmGetResponse(r, esRefID); resource != nil {
+			req.Resources.Apm = append(req.Resources.Apm,
+				resource,
+			)
+		}
+	}
+
+	for _, r := range res.Resources.Appsearch {
+		if resource := parseAppSearchGetResponse(r, esRefID); resource != nil {
+			req.Resources.Appsearch = append(req.Resources.Appsearch,
+				resource,
+			)
+		}
+	}
+
+	for _, r := range res.Resources.EnterpriseSearch {
+		if resource := parseEnterpriseSearchGetResponse(r, esRefID); resource != nil {
+			req.Resources.EnterpriseSearch = append(req.Resources.EnterpriseSearch,
+				resource,
+			)
+		}
+	}
+
+	return &req
+}
+
+func parseElasticsearchGetResponse(r *models.ElasticsearchResourceInfo, esRefID *string) *models.ElasticsearchPayload {
+	plan := r.Info.PlanInfo.Current
+	if plan == nil || plan.Plan == nil {
+		return nil
+	}
+
+	*esRefID = *r.RefID
```

Comment:
The way this variable is being set and passed around is somewhat difficult to understand. I think it would make it more readable to return it alongside `*models.ElasticsearchPayload`. What do you think?